src/node_file.cc

```diff
@@ -882,6 +888,62 @@ static void MKDir(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+
+static size_t CountSlashes(const char* str, size_t len) {
+  size_t cntr = 0;
+  for (size_t i = 0; i < len; i++) {
+    if (str[i] == '/') cntr++;
+  }
+  return cntr;
+}
+
+
+static int ResolveRealPath(Environment* env,
+                           std::string* ret_str,
+                           const char* path) {
+  fs_req_wrap req_wrap;
+  SYNC_CALL_NO_THROW(req_wrap, realpath, path, path);
+
+  if (err != UV_ELOOP) {
+    if (err) return err;
+    *ret_str = std::string(static_cast<const char*>(req_wrap.req.ptr));
+    return err;
+  }
+
+  std::string str_path(path);
+  size_t offset = 0;
+  size_t current = 0;
+  size_t slash_count = 0;
+
+  while ((offset = str_path.find('/', offset + 1)) != std::string::npos) {
+    if (++slash_count < 32) {
+      continue;
+    }
+
+    std::string partial = *ret_str + str_path.substr(current, offset - current);
+    // Call ResolveRealPath() again.
+    int err2 = ResolveRealPath(env, ret_str, partial.c_str());
+    // If a second call deep gets the error I don't much care.
+    if (err2) {
+      *ret_str = std::string();
+      return err2;
+    }
+
+    current = offset;
+    slash_count = CountSlashes(ret_str->c_str(), ret_str->length() - 1);
+  }
+
+  if (offset == std::string::npos) {
+    offset = str_path.length();
+  }
+  if (current < offset) {
+    const char* subs = str_path.substr(current, offset - current).c_str();
```

Comment:
Ah... quick testing shows that it's segfaulting because the suggested change https://github.com/nodejs/node/pull/7548#discussion_r69632975 causes `ResolveRealPath` to recurse indefinitely in some cases. Would need to track it down more but need to get onto a couple other things at the moment.